resources/providers/awslib/multi_region.go

```diff
@@ -92,15 +93,50 @@ func MultiRegionFetch[T any, K any](ctx context.Context, set map[string]T, fetch
 			}
 
 			mux.Lock()
+			defer mux.Unlock()
+			// K might be an slice, struct or pointer
+			// since slice is not comparable we cant just do
+			// var zero K; results == zero
+			// also we cant check if results == nil
+			if shouldDrop(results) {
+				return
+			}
 			crossRegionResources = append(crossRegionResources, results)
-			mux.Unlock()
 		}(client, region)
 	}
 
 	wg.Wait()
 	return crossRegionResources, err
 }
 
+// shouldDrop checks the target type and return true if
+// type is pointer -> the pointer is nil
+// type is struct -> the type zero value
+// type is slice -> len(slice) == 0
+// and false otherwise
+func shouldDrop(t interface{}) bool {
```

Comment:
Cool, why we needed this now?

Reply:
The fetcher can return a slice, struct, or pointer.
When a pointer is nil it will add to the final slice and at the end will have a slice of nils.
To avoid this I thought about checking this.
When we use generics we cant just compare to nil, or check zero value.
So we need to use reflection.